iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/variant_internal.hpp

```diff
@@ -130,23 +136,27 @@ struct call_at_index
         }
     }
 
-    static void copy(const uint64_t index, byte_t* source, byte_t* destination) noexcept
+    static void copy(const uint64_t index, const byte_t* source, byte_t* destination) noexcept
```

Comment:
This could even become `const byte* const source`. We could even add the `restrict` keyword to enable some optimizations if we are sure that the two pointer do not alias the same memory.

Reply:
@elfenpiff restrict only make sense if it is used for both pointer. This tells the compiler that one guarantees that the pointer do not alias the same memory. See also the memcpy documentation.

Reply:
Ok, for now. Let's tackle this if some AUTOSAR warnings catch it or once we make `variant` `constexpr`.